main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -2940,6 +2939,41 @@ public static void removeFromList(final Collection<Geocache> caches, final int o
         }
     }
 
+    public static void moveToList(final Collection<Geocache> caches, final int listId) {
+        if (caches.isEmpty()) {
+            Log.e("No cache was passed to moveToList");
+            return;
+        }
+        final AbstractList list = AbstractList.getListById(listId);
+        if (list == null) {
+            Log.e("Target list with id " + list + " does not exist in moveToList");
+            return;
+        }
+        if (!list.isConcrete()) {
+            return;
+        }
+        init();
+
+        final SQLiteStatement add = PreparedStatement.ADD_TO_LIST.getStatement();
+        final SQLiteStatement remove = PreparedStatement.REMOVE_FROM_ALL_LISTS.getStatement();
+
+        database.beginTransaction();
+        try {
+            for (final Geocache cache : caches) {
+                remove.bindString(1, cache.getGeocode());
+                remove.execute();
+                add.bindLong(1, listId);
+                add.bindString(2, cache.getGeocode());
+                add.execute();
+
+                cache.getLists().add(listId);
```

Comment:
The existing `moveToList()` does a `cache.getLists().remove(oldListId);` now you want to remove all old lists, that would be equivalent to `clear()`, wouldn't it?